test/ex_machina/ecto_test.exs

```diff
@@ -33,19 +73,30 @@ defmodule ExMachina.EctoTest do
       }
     end
 
+    def factory(:article_with_comments) do
+      %MyApp.Article{
+        title: "My Awesome Article with comments",
+        comments: [build(:comment, inverse: true)]
+      }
+    end
+
     def factory(:article, attrs) do
-      %{
-        id: 1,
+      %MyApp.Article{
         title: "My Awesome Article",
         author_id: assoc(attrs, :author, factory: :user).id
       }
     end
 
+    # use `inverse: true` to avoid creating article assoc when using comment
+    # factory in `has_many` relation
     def factory(:comment, attrs) do
-      %{
+      comment = %MyApp.Comment{
         body: "This is great!",
-        article_id: assoc(attrs, :article).id
       }
+      case attrs[:inverse] do
```

Comment:
I hadn't thought of this case before. This could make the API a lot more complicated for people to use. If we can't rethink this to not use extra attrs, we may not want to add this functionality